File: src/main/java/com/spagnola/lcars/panels/LoginPanel.java
package com.spagnola.lcars.panels;

import com.spagnola.lcars.LCARS;
import com.spagnola.lcars.elements.LCARSCorner;
import com.spagnola.lcars.elements.LCARSPanel;

import java.awt.event.ActionListener;

public class LoginPanel extends LCARSPanel {


    /**
     *
     */
    private static final long serialVersionUID = -4009903437795142762L;


    /**
     * @param parent
     * @param loginAction
     */
    public LoginPanel(ActionListener loginAction) {
        super();

        this.setTitle("Login Title");

        LCARSCorner header = new LCARSCorner(this,
                10, 10, 1490, 200,
                75, 400,
                LCARS.EC_WHITE | LCARS.ES_SHAPE_NW | LCARS.ES_LABEL_N | LCARS.ES_STATIC);
        header.setName("Login Header");
        header.setText(header.getName());
        //header.setActionListener(exitActionListener);
        //this.add(header);

        LCARSCorner footerLeft = new LCARSCorner(this,
                10, 980, 450, 100,
                40, 400,
                LCARS.EC_WHITE | LCARS.ES_SHAPE_SW | LCARS.ES_LABEL_NW | LCARS.ES_STATIC);
        footerLeft.setName("Login Footer Left");
        footerLeft.setText(footerLeft.getName());
        //this.add(footerLeft);


    }


}

File: src/main/kotlin/de/visualdigits/lcars/panels/LoginPanel.kt
package de.visualdigits.lcars.panels

import de.visualdigits.lcars.LCARS
import de.visualdigits.lcars.elements.LCARSCorner
import de.visualdigits.lcars.elements.LCARSPanel

class LoginPanel : LCARSPanel() {

    /**
     * @param parent
     * @param loginAction
     */
    init {
        this.setTitle("Login Title")

        val header: LCARSCorner = LCARSCorner(
            x = 10, y = 10, w = 1490, h = 200,
            barH = 75, barV = 400,
            style = LCARS.EC_WHITE or LCARS.ES_SHAPE_NW or LCARS.ES_LABEL_N or LCARS.ES_STATIC
        )
        header.setName("Login Header")
        header.componentText = header.getName()

        //header.actionListener = ActionListener(exitActionListener);
        //this.add(header);
        val footerLeft = LCARSCorner(
            x = 10, y = 980, w = 450, h = 100,
            barH = 40, barV = 400,
            style = LCARS.EC_WHITE or LCARS.ES_SHAPE_SW or LCARS.ES_LABEL_NW or LCARS.ES_STATIC
        )
        footerLeft.setName("Login Footer Left")
        footerLeft.componentText = footerLeft.getName()


        //this.add(footerLeft);
    }

    companion object {
        /**
         *
         */
        private const val serialVersionUID = -4009903437795142762L
    }
}
